test/integration/cni-egress/pod_egress_test.go

```diff
@@ -0,0 +1,223 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cni_egress
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/apps/v1"
+	coreV1 "k8s.io/api/core/v1"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/integration/common"
+)
+
+const (
+	cmdForPodEgressIpv6       = `ip -f inet6 address show dev v6if0 | sed -n "s/.*inet6 \(fd00::ac:.*\)\/118.*/\1/p"`
+	cmdForPodEgressIpv4       = `ip -f inet address show dev v4if0 | sed -n "s/.*inet \(169.254..*\)\/22.*/\1/p"`
+	PublicUrlForEgressTesting = "icanhazip.com"
+)
+
+var _ = Describe("[CANARY] test cluster egress connectivity", func() {
+	var (
+		err error
+		// test container that verifies connectivity to en external IPv6 using IPv6 only
+		testerContainer coreV1.Container
+
+		// Primary node busybox deployment
+		primaryNodeDeployment *v1.Deployment
+
+		// var used for v4 egress
```

Comment:
nit: should we add `v4` or `v6` to the name to distinguish these variables?

Reply:
I introduced a struct to contain v4 and v6 vars to make it clear which var is used for which type of cluster.
```

type v4EgressVars struct {
	pods    v1.PodList
	podsIPs []string
}

// var used for v6 egress
type v6EgressVars struct {
	podsInPrimaryENI      []v1.Pod
	podsInPrimaryENIIPs   []string
	podsInSecondaryENI    []v1.Pod
	podsInSecondaryENIIPs []string
}
```